crates/wasmtime/src/runtime/vm/sys/windows/mmap.rs

```diff
@@ -133,7 +134,13 @@ impl Mmap {
             // Protect the entire file as PAGE_WRITECOPY to start (i.e.
             // remove the execute bit)
             let mut old = 0;
-            if VirtualProtect(ret.as_mut_ptr().cast(), ret.len(), PAGE_WRITECOPY, &mut old) == 0 {
+            if VirtualProtect(
+                ret.as_send_sync_ptr().as_ptr().cast(),
+                ret.len(),
+                PAGE_WRITECOPY,
+                &mut old,
+            ) == 0
+            {
```

Comment:
Changed some of this code's pointer typology (though not the arithmetic) -- would be worth a look.